FungibleToken.ts

```diff
@@ -45,29 +47,30 @@ export class FungibleToken extends TokenContract implements FungibleTokenLike {
   async deploy(props: FungibleTokenDeployProps) {
     await super.deploy(props)
 
-    this.owner.set(props.owner)
+    this.admin.set(props.admin)
     this.supply.set(props.supply)
     this.circulating.set(UInt64.from(0))
 
     this.account.tokenSymbol.set(props.symbol)
     this.account.zkappUri.set(props.src)
   }
 
-  private ensureOwnerSignature() {
-    const owner = this.owner.getAndRequireEquals()
-    return AccountUpdate.createSigned(owner)
+  public getAdminContract(): FungibleTokenAdmin {
+    return (new FungibleTokenAdmin(this.admin.getAndRequireEquals()))
   }
 
   @method
-  async setOwner(owner: PublicKey) {
-    this.ensureOwnerSignature()
-    this.owner.set(owner)
-    this.emitEvent("SetOwner", owner)
+  async setAdmin(admin: PublicKey) {
+    await this.getAdminContract().canAdmin(AdminAction.fromType(AdminAction.types.setAdmin))
```

Comment:
this is not using the `Bool` that `canAdmin()` returns

Reply:
Thanks, good catch!